interface-definitions/include/ospf/protocol-common-config.xml.i

```diff
@@ -621,6 +621,97 @@
     </constraint>
   </properties>
 </leafNode>
+<node name="segment-routing">
+  <properties>
+    <help>Segment-Routing (SPRING) settings</help>
+  </properties>
+  <children>
+    <leafNode name="enable">
```

Comment:
We do not require a dedicated `enable` node. If SR is configured on the CLI we assume the user want's to enable it.

Reply:
Right, so in this case I believe it is needed as a toggle as a distinct on/off for really one reason only. For segment routing normally there are defaults that are there. If someone doesn't want to mess with the defaults they should be able to just enable segment routing. Furthermore, if one doesn't want to change the SRGB/SRLB and wants to use the router as a transit LSR and doesn't want to inject prefixes then they can. That's why I believe in FRR there is a toggle for "on." 

In this case, ISIS and OSPF in FRR both have "segment-routing on" as can be seen:

```
vyos(config)# router isis test
vyos(config-router)# segment-routing
  global-block  Segment Routing Global Block label range
  node-msd      Maximum Stack Depth for this router
  on            Enable Segment Routing
  prefix        Prefix SID
  
vyos(config-router)# exit
vyos(config)# router ospf
vyos(config-router)# segment-routing
  global-block  Segment Routing Global Block label range
  node-msd      Maximum Stack Depth for this router
  on            Enable Segment Routing
  prefix        Prefix SID
```

That's the only reason why I generally use the word "enable" in there. It offers a functionality that doesn't have any other command for.

Reply:
I have made it to now that the "segment-routing" node just needs to be there and it will enable segment-routing within ISIS and OSPF.